src/nvim/path.c

```diff
@@ -1261,36 +1244,39 @@ int gen_expand_wildcards(int num_pat, char **pat, int *num_file, char ***file, i
         return FAIL;
       }
     } else {
+#ifndef UNIX
       // First expand environment variables, "~/" and "~user/".
       if ((has_env_var(p) && !(flags & EW_NOTENV)) || *p == '~') {
         p = expand_env_save_opt(p, true);
         if (p == NULL) {
           p = (char_u *)pat[i];
         }
-#ifdef UNIX
-        /*
-         * On Unix, if expand_env() can't expand an environment
-         * variable, use the shell to do that.  Discard previously
-         * found file names and start all over again.
-         */
-        else if (has_env_var(p) || *p == '~') {
+      }
+#else
+      // First expand environment variables, "~/" and "~user/".
+      if ((has_env_var(p) && !(flags & EW_NOTENV)) || *p == '~') {
+        p = expand_env_save_opt(p, true);
+        if (p == NULL) {
+          p = (char_u *)pat[i];
+        } else if (has_env_var(p) || *p == '~') {
```

Comment:
How about adding a brace after this `else`, and moving the `if` into the block, and putting `#ifdef UNIX` around it.